src/out.rs

```diff
@@ -24,11 +24,22 @@ pub trait Out: io::Write {
     }
 
     /// Ask the user a question, and read in a line with the user's response. If
-    /// there's no user able to respond, `None` will be returned instead.
+    /// there's no user able to respond, an error will be returned instead.
     fn read_line_with_prompt(&mut self, _prompt: &str) -> io::Result<String> {
         Err(io::ErrorKind::Unsupported.into())
     }
 
+    /// Like `read_line_with_prompt`, except will be run asynchronously when
+    /// possible in order to avoid blocking other futures running on the current
+    /// thread. This can be used to allow running background tasks, such as
+    /// fetching resources, while waiting for the user to respond.
+    fn read_line_with_prompt_async<'a>(
+        &'a mut self,
+        initial: &'a str,
+    ) -> Pin<Box<dyn Future<Output = io::Result<String>> + 'a>> {
+        Box::pin(async { self.read_line_with_prompt(initial) })
+    }
```

Comment:
neat

Reply:
This doesn't actually make it async, it's 100% a cheat for cases where we can't make it async